dev-tools/omdb/src/bin/omdb/mgs.rs

```diff
@@ -0,0 +1,499 @@
+                        ..
+                    } => "Gimlet".to_string(),
+                    SpIgnition::Yes {
+                        id: SpIgnitionSystemType::Sidecar,
+                        ..
+                    } => "Sidecar".to_string(),
+                    SpIgnition::Yes {
+                        id: SpIgnitionSystemType::Psc, ..
+                    } => "PSC".to_string(),
+                    SpIgnition::Yes {
+                        id: SpIgnitionSystemType::Unknown(v),
+                        ..
+                    } => format!("unknown: type {}", v),
+                },
+            }
+        }
+    }
+
+    let table_rows = sp_list_ignition.iter().map(IgnitionRow::from);
+    let table = tabled::Table::new(table_rows)
+        .with(tabled::settings::Style::empty())
+        .with(tabled::settings::Padding::new(0, 1, 0, 0))
+        .to_string();
+    println!("{}", textwrap::indent(&table.to_string(), "    "));
+    Ok(())
+}
+
+fn show_sp_states(
+    sp_states: &[(SpIdentifier, SpState)],
+) -> Result<(), anyhow::Error> {
+    #[derive(Tabled)]
+    #[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
+    struct SpStateRow<'a> {
+        #[tabled(rename = "TYPE")]
+        type_: &'static str,
+        slot: u32,
+        model: String,
+        serial: String,
+        rev: u32,
+        hubris: &'a str,
+        pwr: &'static str,
+        rot_active: String,
+    }
+
+    impl<'a> From<&'a (SpIdentifier, SpState)> for SpStateRow<'a> {
+        fn from((id, v): &'a (SpIdentifier, SpState)) -> Self {
+            SpStateRow {
+                type_: sp_type_to_str(&id.type_),
+                slot: id.slot,
+                model: v.model.clone(),
+                serial: v.serial_number.clone(),
+                rev: v.revision,
+                hubris: &v.hubris_archive_id,
+                pwr: match v.power_state {
+                    PowerState::A0 => "A0",
+                    PowerState::A1 => "A1",
+                    PowerState::A2 => "A2",
+                },
+                rot_active: match &v.rot {
+                    RotState::CommunicationFailed { message } => {
+                        format!("error: {}", message)
+                    }
+                    RotState::Enabled { active: RotSlot::A, .. } => {
+                        "slot A".to_string()
+                    }
+                    RotState::Enabled { active: RotSlot::B, .. } => {
+                        "slot B".to_string()
+                    }
+                },
+            }
+        }
+    }
+
+    let table_rows = sp_states.iter().map(SpStateRow::from);
+    let table = tabled::Table::new(table_rows)
+        .with(tabled::settings::Style::empty())
+        .with(tabled::settings::Padding::new(0, 1, 0, 0))
+        .to_string();
+    println!("{}", textwrap::indent(&table.to_string(), "    "));
+    Ok(())
+}
+
+const COMPONENTS_WITH_CABOOSES: &'static [&'static str] =
+    &["sp", "rot", "host-boot-flash"];
```

Comment:
`host-boot-flash` does not have a caboose; that's a hubris image thing. Maybe we could shove a caboose (or something with similar information) into the host phase 1 images? Not sure what that would look like.